### PR TITLE
Add DynastyReader as a provider

### PR DESCRIPTION
**Pull request by yanandcoffee**
@Deathspike @XellossMetallium 

As requested in https://github.com/Deathspike/mangarack/issues/47, this PR will allow a series from DynastyReader to be downloaded on the mangarack-cli via their series URL in this format:  `http://dynasty-scans.com/series/<seriesName>`

Some examples that I've tried to be working:
http://dynasty-scans.com/series/silver_gymnasium
http://dynasty-scans.com/series/dear_my_teacher
http://dynasty-scans.com/series/citrus

Let me know if you find any bugs or want me to change anything.

### CODE REVIEW

File: mangarack-component-core/src/providers/dynastyreader/chapter.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/chapter.ts]: Don't use `any`. You're expecting a `{image: string}`, so mention this explicitly. This will also ensure a future reader will understand what it is you're expecting to parse there, and if this changes in the future, the new type cast will safe guard additional code (even if it is pretty obvious in this code, it's a good habit to get into).

File: mangarack-component-core/src/providers/dynastyreader/default.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/default.ts]: This is again a particular weirdness of mine, could you make the return into an `else` statement? Right now, you really have to read the throw statement to see the function can exit there, while an else would make it immediately clear the function can stop there. This merely improves readability, and I particularly love readability.

File: mangarack-component-core/src/providers/dynastyreader/page.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/page.ts]: What's this blank line for?

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: What's this blank line for?

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: I'm pretty sure this remap is incorrect. The idea is to take the definitions of the provider (dynasty reader) and remap them into the `GenreType` enumerator (so mangarack understands it). If the name matches, great, no remap is needed, but when it doesn't, a remap is needed. Sci-fi on mangafox, for example, remaps to Science Fiction (as per the enumerator). Now, you're remapping Adult to NSFW here, which is a genre type that does not exist in mangarack. Please check the full list of genres of dynasty reader and compare them with the GenreType enumerator.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line above, and like earlier, wrap the throw in an else for readability.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Blank line.

File: mangarack-component-core/src/providers/dynastyreader/series.ts
Comment (Deathspike) [inline mangarack-component-core/src/providers/dynastyreader/series.ts]: Like with the throws, making these explicit blocks make it very clear at one glance you've got 3 exits of this function. So, if, else if, else. Again, readability weirdness of mine.

**Review by Deathspike — CHANGES_REQUESTED** [review]
Awesome pull merge! I've added some feedback for changes/suggestions, all regarding to code style. Loved the usage of `const`, I will have to pick this up for the rest of the code base too! I will be happy to clarify/discuss comments if needed, and will take it all for a test run/merge once done 👍 Sorry for the delay!

File: mangarack-component-core/src/providers/dynastyreader/default.ts
Comment (yanandcoffee) [inline mangarack-component-core/src/providers/dynastyreader/default.ts]: A `throw` or a `return` will always stop and exit the function, so once a `throw` is entered, it's implicit that it will exit the function. It is also common practice to guard classes (conditionals) so that the function can exit without going further, as it has been done in other areas of the `mangarack` code. I'm not sure putting an `else` actually improves readability; because this can lead to "conditional hell", but it is really preference so I'll respect whatever code style you prefer.

**Review by yanandcoffee — COMMENTED** [review]
Changed code to be more consistent with project coding style.

**Comment by yanandcoffee** [comment]
@Deathspike thanks for the comments, I've made changes accordingly and left some comments back. Should be good to go once you confirm the genre type mapping and that the code works as intended.

**Comment by Deathspike** [comment]
Sorry for the late reply!

I will be testing and merging this in 4.1.0. I hope to get this new version out in 1~2 weeks.

Note to self: Check the remapping dictionaries.

**Comment by yanandcoffee** [comment]
@Deathspike didn't check on this until now, but just FYI, I merged updated master into this fork so there's no more conflicts. Let me know if you need more changes to be done for this to be merged.

**Pull request closed without merging**
